src/main/java/com/example/springboot/dao/BookDAOImpl.java

```diff
@@ -0,0 +1,181 @@
+package com.example.springboot.dao;
+
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+
+import org.apache.log4j.Logger;
+import org.springframework.stereotype.Repository;
+
+import com.example.springboot.base.Book;
+import com.mysql.jdbc.Statement;
+
+/**
+ * This class implement methods from BookDAO interface
+ * @author Dmytro Kohut
+ * @version 1.1
+ */
+@Repository("bookDAO")
+public class BookDAOImpl implements DAOInterface<Book> {
+	
+	final static Logger LOGGER = Logger.getLogger(BookDAOImpl.class);
+	
+	static {
+		try {
+			Class.forName("com.mysql.jdbc.Driver").newInstance();
+		} catch(InstantiationException | IllegalAccessException | ClassNotFoundException e) {
+			LOGGER.error("ERROR ", e);
+		}
+	}
+	
+	/**
+	 * This method create and return a connection
+	 * @return
+	 * @throws SQLException
+	 */
+	private Connection getConnection() throws SQLException {
+		return DriverManager.getConnection("jdbc:mysql://localhost:3306/libriary", "root", "root");
+	}
+	
+	/**
+	 * This method close an given connection
+	 * @param connection
+	 */
+	private void closeConnection(Connection connection) {
+		try {
+			connection.close();
+		} catch(SQLException e) {
+			LOGGER.error("ERROR ", e);
+		}
+	}
+	
+	/**
+	 * @see com.example.springboot.dao.DAOInterface#findAllBooks()
+	 */
+	@Override
+	public Collection<Book> findAll() {
+		Connection connection = null;
+		List<Book> resultList = new ArrayList<>();
+		
+		try {
+			connection = getConnection();
+			PreparedStatement statement = connection.prepareStatement("SELECT * FROM `books`;");
```

Comment:
Move query into static variable

Reply:
On the query you HAVE TO use list of fields, that you are trying to fetch not `*`